83/transform cbbr data

pratishta opened the pull request:
Closes https://github.com/NYCPlanning/ae-data-flow/issues/83

Notes:
- Currently manually created the model tables so that the transform can update them. The code to create these tables will eventually be deleted and done in drizzle on the zoning-api repo
- Decided to make two tables: `community_board_budget_request` and `agency_response`
- I did not include multiple columns from cbbr_export like intersection, supporters, address. The reason is that the values for these columns are either inconsistent in content or across the records. I also do not see what value the address / intersection / cross street columns would provide that the geom would not
- I manually coerced some agencies based on their managing code 
- Removed the code for creating `flow_borough` table since we confirmed that it's not used anywhere and is likely extraneous

TangoYankee reviewed (changes requested):
I agree with only adding the fields that are straightforward to manage and would have a high impact. 

For this first review, I focused on the `types` and `constraints` of the fields on the model tables. I did quickly look at the transformations and they initially look good. However, I expect type and constraint updates to cause updates to the transformations

TangoYankee commented on pg/model-transform/community-board-budget-requests.sql:
```diff
@@ -77,8 +79,89 @@ WHERE
 COPY cbbr_need_group TO '/var/lib/postgresql/data/cbbr_need_group.csv';
 COPY cbbr_need TO '/var/lib/postgresql/data/cbbr_need.csv';
 COPY cbbr_request TO '/var/lib/postgresql/data/cbbr_request.csv';
 COPY cbbr_option_cascade TO '/var/lib/postgresql/data/cbbr_option_cascade.csv';
```
Do we also need to export `cbbr_agency_category_response`?

TangoYankee commented on db/pg/model-create/community-board-budget-requests.sql:
```diff
@@ -9,28 +9,36 @@ DROP TABLE IF EXISTS
+	"need_id" smallint NOT NULL,
+	"request_id" smallint NOT NULL,
+	"explanation" text NOT NULL,
+	"is_location_specific" text NOT NULL,
```
should this be `text` or `boolean`?

TangoYankee commented on db/pg/model-create/community-board-budget-requests.sql:
```diff
@@ -9,28 +9,36 @@ DROP TABLE IF EXISTS
+	"request_id" smallint NOT NULL,
+	"explanation" text NOT NULL,
+	"is_location_specific" text NOT NULL,
+	"geom" geometry(geometryCollection,4326)
```
What are thoughts around using `geometryCollection`?

pratishta replied:
Initially, I wanted it to be one column that had all the geospatial data, but after a short search and looking at our existing tables, I realized that it's much better to separate the different data types 🫡

pratishta commented:
I updated the PR with the feedback from @TangoYankee in addition to the following

- I went ahead and defined the schema in zoning-api [here](https://github.com/NYCPlanning/ae-zoning-api/pull/492) and then generated the migration file, which I then put in `cbbr-temp-model-create.sql` (this runs last in the shell script which makes it easier to test)
- Had to manually add managing code 859 which in cbbr_export shows as MOME's code. In the source_agency MOME's code shows up as NULL. My feeling is that if 859 does in fact correspond with MOME, it should be source_agency so that we don't have to manually add it 
- To transform the cbbr_export data, I added columns to the source table and then directly added from those columns to the model table. I wanted to avoid updating the original source_cbbr_export columns and make the insert into the model table straightforward 
- Added the lift and mercator columns for points and polygons each. I didn't add a label column because 1) I noticed that we don't use the capital projects one and 2) I don't see where in the designs we might want to label. It seems like that column is only important to labeling the admin boundaries
- Removed `flow_borough` since we confirmed we don't use it anywhere 
- I encountered some clean up necessary in the source data. E.g. Trimming whitespace, left padding with 0 for agency codes, mismatch in case between foreign keys which requires me to compare with UPPER(), etc. I'm concerned with the consistency in the quality of data we're getting and wondering if there's a wholesale formatting we should do ourselves in this data flow step...

pratishta commented on db/pg/model-create/community-board-budget-requests.sql:
```diff
@@ -7,5 +7,30 @@ DROP TABLE IF EXISTS
+
+CREATE TABLE IF NOT EXISTS community_board_budget_request (
+    "id" text NOT NULL,
+    "tracking_code" text NOT NULL,
```
I realized I didn't assign a primary key. Is there a reason you're looking at `tracking_code` over `id`?

pratishta commented on pg/model-transform/community-board-budget-requests.sql:
```diff
@@ -77,8 +79,124 @@ WHERE
+	mercator_fill_m_pnt,
+	mercator_fill_m_poly
+)
+SELECT
```
I think I need to include a `DISTINCT` here as I notice that there are duplicates in the `community-board-budget-request` table
